### PR TITLE
Fixes menu_link_item entity visibility issue.

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-engagement.atlassian.net/browse/SDPA-4589

### Issue
The Drupal menu API endpoint returns menu items the user shouldn't have access to

### Changes
1. implements `hook_ENTITY_TYPE_access` to check if the node that been attached to the menu published or not.

### CODE REVIEW

File: tide_core.module [hunk 1]
```diff
@@ -244,3 +247,20 @@ function tide_core_editor_js_settings_alter(array &$settings) {
...
+function tide_core_menu_link_content_access(EntityInterface $menu_link_item, $operation, AccountInterface $account) {
+  $url = $menu_link_item->getUrlObject();
+  if (!$url->isExternal() && $url->isRouted() && $menu_link_item->isPublished()) {
+    $parameters = $url->getRouteParameters();
+    if (isset($parameters['node'])) {
+      $node = Node::load($parameters['node']);
+      if ($account->hasPermission('administer menu')) {
+        return AccessResult::neutral();
+      }
+      return $node->isPublished() ? AccessResult::neutral() : AccessResult::forbidden();
```
Comment (MdNadimHossain) [inline tide_core.module:263]: isn't it going to return the `AccessResult::neutral()` if the node is published? What if it does not has the permission `administer menu` and it is published?

**Review by MdNadimHossain — COMMENTED** [review]
@vincent-gao general question. `$menu_link_item->isPublished` does it check if the node is published, I mean the menu link item is node right!

File: tide_core.module (hunk 1 above)
Comment (vincent-gao) [reply tide_core.module:263]: this is a good point.

**Comment by vincent-gao** [comment]
> @vincent-gao general question. `$menu_link_item->isPublished` does it check if the node is published, I mean the menu link item is node right!

menu link is not node.

**Review by MdNadimHossain — APPROVED** [review]

**Pull request closed without merging**